tests/scalers/loki/loki_test.go

```diff
@@ -0,0 +1,262 @@
+func TestLokiScaler(t *testing.T) {
+	// Create kubernetes resources
+	kc := GetKubernetesClient(t)
+	InstallLoki(t, testNamespace)
+
+	// Create kubernetes resources for testing
+	data, templates := getTemplateData()
+	KubectlApplyMultipleWithTemplate(t, data, templates)
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, monitoredAppName, testNamespace, 1, 60, 3),
+		"replica count should be %d after 3 minutes", minReplicaCount)
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minReplicaCount, 60, 3),
+		"replica count should be %d after 3 minutes", minReplicaCount)
+
+	testActivation(t, kc, data)
+	testScaleUp(t, kc, data)
+	testScaleDown(t, kc)
+
+	// cleanup
+	KubectlDeleteMultipleWithTemplate(t, data, templates)
+	UninstallLoki(t, kc, testNamespace)
+}
+
+func testActivation(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- testing activation ---")
+	KubectlApplyWithTemplate(t, data, "generateLowLevelLoadJobTemplate", generateLowLevelLoadJobTemplate)
+
+	AssertReplicaCountNotChangeDuringTimePeriod(t, kc, deploymentName, testNamespace, minReplicaCount, 60)
+}
+
+func testScaleUp(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- testing scale up ---")
+	KubectlApplyWithTemplate(t, data, "generateLoadJobTemplate", generateLoadJobTemplate)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, maxReplicaCount, 60, 3),
+		"replica count should be %d after 3 minutes", maxReplicaCount)
+}
+
+func testScaleDown(t *testing.T, kc *kubernetes.Clientset) {
+	t.Log("--- testing scale down ---")
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minReplicaCount, 60, 5),
+		"replica count should be %d after 5 minutes", minReplicaCount)
+}
+
+func getTemplateData() (templateData, []Template) {
+	return templateData{
+			TestNamespace:         testNamespace,
+			DeploymentName:        deploymentName,
+			PublishDeploymentName: publishDeploymentName,
+			ScaledObjectName:      scaledObjectName,
+			MonitoredAppName:      monitoredAppName,
+			LokiServerName:        lokiServerName,
+			MinReplicaCount:       minReplicaCount,
+			MaxReplicaCount:       maxReplicaCount,
+		}, []Template{
+			{Name: "deploymentTemplate", Config: deploymentTemplate},
+			{Name: "monitoredAppDeploymentTemplate", Config: monitoredAppDeploymentTemplate},
+			{Name: "scaledObjectTemplate", Config: scaledObjectTemplate},
+		}
+}
+
+func InstallLoki(t *testing.T, namespace string) {
+	_, err := ExecuteCommand("helm repo add grafana https://grafana.github.io/helm-charts")
+	assert.NoErrorf(t, err, "cannot execute command - %s", err)
+	_, err = ExecuteCommand("helm repo update")
+	assert.NoErrorf(t, err, "cannot execute command - %s", err)
+	_, err = ExecuteCommand(fmt.Sprintf("helm upgrade --install loki grafana/loki-stack --wait --namespace=%s --create-namespace", namespace))
```

Comment:
don't create the namespace here. Every namespace you need to create has to be created with the `CreateNamespace` helper because we annotate the e2e namespaces with a specific label that we use during the general cleanup to ensure that there isn't any e2e resource after finishing the e2e tests.
In this case, you should create the namespace manually before calling `InstallLoki` and this line should be like
```suggestion
	_, err = ExecuteCommand(fmt.Sprintf("helm upgrade --install loki grafana/loki-stack --wait --namespace=%s", namespace))
```